Lib/idlelib/browser.py

```diff
@@ -19,14 +19,45 @@
 from idlelib.tree import TreeNode, TreeItem, ScrolledCanvas
 from idlelib.windows import ListedToplevel
 
+
 file_open = None  # Method...Item and Class...Item use this.
 # Normally pyshell.flist.open, but there is no pyshell.flist for htest.
 
+
+def _traverse_node(node, name=None):
+    """Return the immediate children for a node.
+
+    Node is the current node being traversed.  The return value is
+    a tuple with the first value being a dictionary of the
+    Class/Function instances of the node and the second is
+    a list of tuples with (lineno, name).
+    """
+    items = []
+    children = {}
+    for key, cl in node.items():
+        if name is None or cl.module == name:
+            s = key
```

Comment:
Abbreviating 'key' as 's' seems pointless and confusing.  (I know, you copied it.)  Unless you see something I don't, delete and use 'key' below.

Reply:
I wasn't sure how much you wanted to refactor before knowing the tests were good.